src/components/BuyPremium.tsx

```diff
@@ -0,0 +1,30 @@
+"use client";
+
+import clsx from "clsx";
+import { shallow } from "zustand/shallow";
+import PremiumBadge from "~/components/PremiumBadge";
+import { MdiCrown } from "~/components/icons";
+import { useTranslation } from "~/i18n/client";
+import { useAppStore } from "~/stores";
+
+const isMobile = process.env.NEXT_PUBLIC_DEPLOY_ENV === "mobile";
+
+export default function BuyPremium({ className }: { className?: string }) {
+  const { t } = useTranslation();
+  const [premium, setOpen] = useAppStore(
+    ({ config, ui }) => [config.packages, ui.setShowInAppPurchasesDialog],
```

Comment:
```suggestion
    ({ config, ui }) => [config.premium, ui.setShowInAppPurchasesDialog],
```
shouldn't it be config.premium here?